### DISCUSSION TITLE
CuTe and Tensor Core

### DISCUSSION THREAD

**Discussion by osayamenja** [NVIDIA/cutlass#1619] [Q&A] [answered]
This may be a naive question, so please feel free to direct me to documentation or code that resolves it.

## TL;DR 

Does CuTe `gemm` with TiledMMA leverage the tensor core?

## MMA Instructions
[Per](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#warp-matrix-functions) the CUDA C++ programming guide, `wmma` targets the tensor core via warp level MMA.

Also, the PTX ISA [mentions](https://docs.nvidia.com/cuda/parallel-thread-execution/#warp-level-matrix-instructions) both `wmma` and `mma` under warp-level MMA but does not _explicitly_ say that the latter instruction targets tensor cores.

CUTLASS uses `wmma` and `mma` for the `cutlass` namespace (see [here](https://github.com/NVIDIA/cutlass/tree/637b15906358191cb4238af419d408a65819d7ec/include/cutlass/arch)) and only `mma` for `cute` (see [here](https://github.com/NVIDIA/cutlass/blob/637b15906358191cb4238af419d408a65819d7ec/include/cute/arch/mma_sm80.hpp#L74)), leading to the question I asked above. 

Also, tensor core [examples](https://github.com/NVIDIA/cutlass/blob/637b15906358191cb4238af419d408a65819d7ec/examples/19_tensorop_canonical/tensorop_canonical.cu) available use `cutlass` only and there are none using `cute`, so my understanding is that maybe this is the case because there is no support?

Please help me clarify.

**Comment by thakkarV** [comment, marked as answer]
Yes. Using the mma ptx targets tensor cores.

**Reply by osayamenja** [reply]
Thank you! @thakkarV
